### PR TITLE
Pull stream start improvements

### PR DESCRIPTION
**Pull request by mjh1**

### CODE REVIEW

**Review by thomshutt — APPROVED** [review]

File: packages/api/src/controllers/helpers.ts [hunk 1]
```diff
@@ -661,12 +661,16 @@ export const triggerCatalystPullStart =
             timeout: PULL_START_TIMEOUT,
             maxRedirects: 10,
           });
-          const body = await res.text();
-          const isHlsErr = body.includes("#EXT-X-ERROR: Stream open failed");
+          const errHeader = res.headers.get("error");
```
Comment (victorges) [inline packages/api/src/controllers/helpers.ts:664]: ah nice

**Review by victorges — APPROVED** [review]
lgtm anyway

**Review by victorges — COMMENTED** [review]
test

File: packages/api/src/controllers/stream.ts [hunk 2]
```diff
@@ -1097,20 +1068,7 @@ app.put(
...
-          );
-        } catch (err) {
-          logger.error("Error triggering catalyst pull start err=", err);
-        }
-      });
-
-      if (waitActive === "true") {
-        stream = await pollWaitStreamActive(req, stream.id);
-      }
+      await triggerCatalystPullStart(stream, getHLSPlaybackUrl(ingest, stream));
```
Comment (victorges) [inline packages/api/src/controllers/stream.ts:1071]: so in practice we always wait for the stream to start, is that right?

Comment (mjh1) [reply packages/api/src/controllers/stream.ts:1071]: yes if the manifest request has returned then the stream has started and ready for playback

**Pull request merged**